src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleasePermissionService.cs

```diff
@@ -0,0 +1,183 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Admin.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using static GovUk.Education.ExploreEducationStatistics.Content.Model.ReleaseRole;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class ReleasePermissionService : IReleasePermissionService
+    {
+        private readonly ContentDbContext _contentDbContext;
+        private readonly IPersistenceHelper<ContentDbContext> _persistenceHelper;
+        private readonly IUserReleaseRoleRepository _userReleaseRoleRepository;
+        private readonly IUserService _userService;
+
+        public ReleasePermissionService(
+            ContentDbContext contentDbContext,
+            IPersistenceHelper<ContentDbContext> persistenceHelper,
+            IUserReleaseRoleRepository userReleaseRoleRepository,
+            IUserService userService)
+        {
+            _contentDbContext = contentDbContext;
+            _persistenceHelper = persistenceHelper;
+            _userReleaseRoleRepository = userReleaseRoleRepository;
+            _userService = userService;
+        }
+
+        public async Task<Either<ActionResult, List<ContributorViewModel>>>
+            ListReleaseContributors(Guid publicationId, Guid releaseId)
+        {
+            return await _persistenceHelper
+                .CheckEntityExists<Publication>(publicationId,
+                    query =>
+                        query.Include(r => r.Releases))
+                .OnSuccessDo(publication => _userService
+                    .CheckCanUpdateReleaseRole(publication, Contributor))
+                .OnSuccess(async publication =>
+                {
+                    var release = publication.Releases.SingleOrDefault(r => r.Id == releaseId);
+
+                    if (release == null)
+                    {
+                        return new Either<ActionResult, List<ContributorViewModel>>(new NotFoundResult());
```

Comment:
Could simplify this to the following (as we have an implicit converter that automatically wraps this with an Either):

```cs
return new NotFoundResult();
```

Probably won't need this though if we just remove the initial check for a publication (see above comment).